source/common/upstream/cluster_manager_impl.h

```diff
@@ -361,6 +362,18 @@ class ClusterManagerImpl : public ClusterManager, Logger::Loggable<Logger::Id::u
   // This map is ordered so that config dumping is consistent.
   typedef std::map<std::string, ClusterDataPtr> ClusterMap;
 
+  struct PendingUpdates {
+    PendingUpdates() {}
+    Event::TimerPtr timer;
+    std::unordered_set<HostSharedPtr> added;
+    std::unordered_set<HostSharedPtr> removed;
```

Comment:
hmm, these should probably be `ordered` so that the merged update sees hosts in the same order it would have seen them if they were delivered immediately?

cc: @zuercher @mattklein123

Reply:
maybe not that important and not worth the cost...

Reply:
to keep it ordered we'd have to keep two sets (added/removed) to check uniqueness (e.g.: that a host isn't added/removed twice) and we'd also have to keep two vectors (added/removed) to preserve the order..... or am I missing an ordered set data structure hidding somewhere? :-)